onnxruntime/python/tools/transformers/fusion_reshape.py

```diff
@@ -31,16 +48,43 @@ def fuse(self, reshape_node, input_name_to_nodes, output_name_to_node):
 
         (unsqueeze_0, gather_0, shape_0) = path0
 
+        shape = []
+        gather_value = self.model.get_constant_value(gather_0.input[1])
+        if gather_value == 0:
+            shape.append(0)
+
         path1 = self.model.match_parent_path(concat_node, ['Unsqueeze', 'Gather', 'Shape'], [1, 0, 0],
                                              output_name_to_node)
         if path1 is None:
+            # Adjust for Bart
+            if len(concat_node.input) == 4:
+                input_1_proto = self.model.get_initializer(concat_node.input[1])
+                input_2_proto = self.model.get_initializer(concat_node.input[2])
+                input_3_proto = self.model.get_initializer(concat_node.input[3])
+                if input_1_proto is None or input_2_proto is None or input_3_proto is None:
+                    return
+                input_1 = numpy_helper.to_array(input_1_proto)
+                input_2 = numpy_helper.to_array(input_2_proto)
+                input_3 = numpy_helper.to_array(input_3_proto)
+                if input_1[0] == -1 and input_2[0] > 0 and input_3[0] > 0:
```

Comment:
also check len(input_1) == 1 and len(input_2) == 1 and len(input_3) == 1?

Reply:
good catch!